cmd/main.go

```diff
@@ -52,14 +54,14 @@ func main() {
 
 	// Gathering the Credentials Handler.
 	headerCredentials := handlers.NewHeaderCredentialsHandler(logger)
-	cookieCredentials := handlers.NewCookieCredentialsHandler(logger)
+	cookieCredentials := handlers.NewCookieCredentialsHandler(logger, authServiceConfiguration)
 	// preference is chosen here.
 	credentials := []handlers.CredentialsHandler{
 		headerCredentials,
 		cookieCredentials,
 	}
 
-	restController := controllers.NewRESTController(logger, restConfiguration, oidcService, credentials)
+	restController := controllers.NewRESTController(logger, restConfiguration, authServiceConfiguration, oidcService, credentials)
```

Comment:
why isn't `RestConfiguration` receiving the configuration object?

Reply:
and i don't think that RestController should be aware of how the cookie is called, that's why we have the handlers, they should perform as middlewares for this and abstract this logic